docs: "Writing a Tendermint Core application in Java (gRPC)" guide (#3887)
tendermint/tendermint: docs/guides/java.md

    var v = getPersistedValue(k);
    var builder = ResponseQuery.newBuilder();
    if (v == null) {
        builder.setLog("does not exist");
    } else {
        builder.setLog("exists");
        builder.setKey(ByteString.copyFrom(k));
        builder.setValue(ByteString.copyFrom(v));
    }
    responseObserver.onNext(builder.build());
    responseObserver.onCompleted();
}
```

The complete specification can be found
[here](https://tendermint.com/docs/spec/abci/).

## 1.4 Starting an application and a Tendermint Core instances

Put the following code into the `$KVSTORE_HOME/src/main/java/io/example/App.java` file:

```java
package io.example;

import jetbrains.exodus.env.Environment;
import jetbrains.exodus.env.Environments;

import java.io.IOException;

public class App {
    public static void main(String[] args) throws IOException, InterruptedException {
        try (Environment env = Environments.newInstance("tmp/storage")) {
            var app = new KVStoreApp(env);
            var server = new GrpcServer(app, 26658);
            server.start();
            server.blockUntilShutdown();
        }
    }
}
```

It is the entry point of the application.

> Here we create special object `Environment` which knows where to store state of the application.

Reviewer: ```suggestion
Here we create a special object `Environment`, which knows where to store the application state.
```

> Then we create and srart gRPC server to handle Tendermint's requests.

Reviewer: ```suggestion
Then we create and start the gRPC server to handle Tendermint Core requests.
```

> Create file `$KVSTORE_HOME/src/main/java/io/example/GrpcServer.java`:

Reviewer: ```suggestion
Create the `$KVSTORE_HOME/src/main/java/io/example/GrpcServer.java` file with the following content:
```
